spyder/utils/introspection/old_fallback.py

```diff
@@ -8,305 +8,306 @@
-        matches = get_matches(patterns, source, token, start_line)
-    # find the one closest to the start line (prefer before the start line)
-    if matches:
-        min_dist = len(source.splitlines())
-        best_ind = 0
-        for match in matches:
-            dist = abs(start_line - match)
-            if match <= start_line or not best_ind:
-                if dist < min_dist:
-                    min_dist = dist
-                    best_ind = match
-    if matches:
-        if DEBUG_EDITOR:
-            log_dt(LOG_FILENAME, 'regex definition match', t0)
-        return best_ind
-    else:
-        if DEBUG_EDITOR:
-            log_dt(LOG_FILENAME, 'regex definition failed match', t0)
-        return None
-
-
-def get_matches(patterns, source, token, start_line):
-    patterns = [pattern.format(token, r'[^0-9a-zA-Z.[]')
-                for pattern in patterns]
-    pattern = re.compile(r'|^'.join(patterns))
-    # add the trailing space to allow some regexes to match
-    lines = [line.strip() + ' ' for line in source.splitlines()]
-    if start_line == -1:
-        start_line = len(lines)
-    matches = []
-    for (index, line) in enumerate(lines):
-        if re.match(pattern, line):
-            matches.append(index + 1)
-    return matches
-
-
-def python_like_exts():
-    """Return a list of all python-like extensions"""
-    exts = []
-    for lang in languages.PYTHON_LIKE_LANGUAGES:
-        exts.extend(list(languages.ALL_LANGUAGES[lang]))
-    return ['.' + ext for ext in exts]
-
-
-def all_editable_exts():
-    """Return a list of all editable extensions"""
-    exts = []
-    for (language, extensions) in languages.ALL_LANGUAGES.items():
-        exts.extend(list(extensions))
-    return ['.' + ext for ext in exts]
-
-
-def _listdir(root):
-    "List directory 'root' appending the path separator to subdirs."
-    res = []
-    root = os.path.expanduser(root)
-    try:
-        for name in os.listdir(root):
-            path = os.path.join(root, name)
-            if os.path.isdir(path):
-                name += os.sep
-            res.append(name)
-    except:
-        pass  # no need to report invalid paths
-    return res
-
-
-def _complete_path(path=None):
-    """Perform completion of filesystem path.
-    https://stackoverflow.com/questions/5637124/tab-completion-in-pythons-raw-input
-    """
-    if not path:
-        return _listdir('.')
-    dirname, rest = os.path.split(path)
-    tmp = dirname if dirname else '.'
-    res = [p for p in _listdir(tmp) if p.startswith(rest)]
-    # more than one match, or single match which does not exist (typo)
-    if len(res) > 1 or not os.path.exists(path):
-        return res
-    # resolved to a single directory, so return list of files below it
-    if os.path.isdir(path):
-        return [p for p in _listdir(path)]
-    # exact file match terminates this completion
-    return [path + ' ']
+# from __future__ import print_function

```

Comment:
@ccordoba12 why did we keep this file?

Could we just delete it? not even the imports makes sense at this point

Reply:
No, please keep it. The code here can be used to provide better support in the editor for Cython files.

Reply:
Ok